src/components/demos/write/Demo.js

```diff
@@ -21,138 +24,122 @@ const Wrapper = styled.div`
       attemptLogin()
     }
-  }, [usernamePasswordPairs, execute, binaryName, theme])
+    // TODO address demoState dependency, gets trapped in re-rendering state
+  }, [bin, execute, update, usernamePasswordPairs])
+
+  useEffect(() => {
+    update(initState)
+  }, [update])
 
   useEffect(() => {
-    if (isMorello && (failedLogin(apiOutput) || loginError(apiOutput))) {
+    if (isMorello && (failedLogin(output) || loginError(output))) {
       setAnimateLoginFailed(true)
       setTimeout(() => setAnimateLoginFailed(false), 1000)
     }
-  }, [apiOutput, isMorello, setAnimateLoginFailed])
+  }, [output, isMorello, setAnimateLoginFailed])
 
   return (
     <>
       <Header {...props} showClose={true} />
       <Wrapper {...theme.wrapper}>
-        {successfulLogin(apiOutput) ? (
-          <SecretDesktop icons={props.secretDesktop} />
+        {successfulLogin(output) ? (
+          <SecretDesktop font={theme.font} icons={props.secretDesktop} />
         ) : (
-          <>
-            <Help
-              theme={theme}
-              content={helpContent}
-              showContentState={showHelp}
-              setShowContentState={setShowHelp}
-            />
-            <Box {...demoState} animate={animateLoginFailed}>
-              <Container
-                styles={{ height: '100%', paddingTop: '150px' }}
-                size={10}
-              >
-                <LoginForm
-                  demoState={demoState}
-                  showSpinner={awaitingApi}
-                  setUsernamePasswordPairs={setUsernamePasswordPairs}
-                  apiOutput={apiOutput}
-                  setApiOutput={setApiOutput}
-                />
-              </Container>
-            </Box>
-          </>
+          <Box {...demoState} animate={animateLoginFailed}>
+            <Container
+              styles={{ height: '100%', paddingTop: '150px' }}
+              size={10}
+            >
+              <LoginForm demoState={demoState} />
+            </Container>
+          </Box>
         )}
         {!isMorello
-          ? successfulLogin(apiOutput) && (
-              <ButtonSide action={switchToMorello} />
-            )
-          : (successfulLogin(apiOutput) || loginError(apiOutput)) && (
+          ? successfulLogin(output) && <ButtonSide action={switchToMorello} />
+          : (successfulLogin(output) || loginError(output)) && [
               <ButtonSide
+                key={'write-demo-side-btn'}
                 message={'Learn More'}
                 action={() => {
                   nav('/write-demo-explainer')
                 }}
-              />
-            )}
+              />,
+              <ConsoleButton
+                key={'write-demo-console-icon'}
+                update={update}
+                state={demoState}
+                font={theme.font}
+              />,
+            ]}
+        <Help theme={theme} content={helpContent} showContentState={showHelp} />
```

Comment:
missing a value for `setShowContentState`. Button doesn't work right now

Reply:
no longer needed, using ctx.